libs/api/ebpf_api.cpp

```diff
@@ -1464,26 +1469,31 @@ _delete_ebpf_object(_In_opt_ _Post_invalid_ ebpf_object_t* object) noexcept
     }
 }
 
-static void
-_remove_ebpf_object_from_globals(_In_ const ebpf_object_t* object) noexcept
+_Requires_lock_not_held_(_ebpf_state_mutex) static void _remove_ebpf_object_from_globals(
+    _In_ const ebpf_object_t* object) noexcept
 {
+    std::unique_lock lock(_ebpf_state_mutex);
```

Comment:
nit(perf): perhaps put this lock after `ebpf_assert(object);` line since you don't need the lock for a log or the assert?